### PR TITLE
Simplify some Option / Result / ? operator patterns

### PR DESCRIPTION
**Pull request by huitseeker**
When those match a combinator exactly.

Tool-aided by [comby-rust](https://github.com/huitseeker/comby-rust).

### CODE REVIEW

File: client/service/src/lib.rs [hunk 1]
```diff
@@ -402,9 +401,7 @@ fn start_rpc_servers<
 						start(&address)
 					},
 					_ => Err(e),
-				})?),
-			None => None,
-		})
+				}) }).transpose()?)
```
Comment (bkchr) [inline client/service/src/lib.rs:404]:
```suggestion
			})).transpose()
```

**Review by bkchr — COMMENTED** [review]
It would be nice if you could check the output of your tool before opening a pr. 

There are still some other extra unneeded `{}`. Please remove them.

File: client/api/src/notifications.rs [hunk 2]
```diff
@@ -53,15 +53,13 @@ impl StorageChangeSet {
 			.iter()
 			.filter_map(move |(sk, changes)| {
 				if let Some(cf) = self.child_filters.as_ref() {
-					if let Some(filter) = cf.get(sk) {
-						Some(changes
+					cf.get(sk).map(|filter| { changes
```
Comment (shawntabrizi) [inline client/api/src/notifications.rs:56]: couldn't this also get turned into a `.map`?

File: client/peerset/src/peersstate.rs [hunk 3]
```diff
@@ -272,15 +272,11 @@ impl PeersState {
 			})
 			.map(|(peer_id, _)| peer_id.clone());
 
-		if let Some(peer_id) = outcome {
-			Some(NotConnectedPeer {
+		outcome.map(move |peer_id| { NotConnectedPeer {
```
Comment (shawntabrizi) [inline client/peerset/src/peersstate.rs:275]:
```suggestion
		outcome.map(move |peer_id| NotConnectedPeer {
```

File: client/peerset/src/peersstate.rs [hunk 4]
```diff
@@ -272,15 +272,11 @@ impl PeersState {
...
 				state: self,
 				set,
 				peer_id: Cow::Owned(peer_id),
-			})
-		} else {
-			None
-		}
+			} })
```
Comment (shawntabrizi) [inline client/peerset/src/peersstate.rs:279]:
```suggestion
			})
```

File: client/rpc/src/state/state_light.rs [hunk 5]
```diff
@@ -722,13 +722,10 @@ fn maybe_share_remote_request<Block: BlockT, Requests, V, IssueRequest, IssueReq
 fn display_error<F, T>(future: F) -> impl std::future::Future<Output=Result<T, ()>> where
 	F: std::future::Future<Output=Result<T, Error>>
 {
-	future.then(|result| ready(match result {
-		Ok(result) => Ok(result),
-		Err(err) => {
+	future.then(|result| ready(result.or_else(|err| { {
```
Comment (shawntabrizi) [inline client/rpc/src/state/state_light.rs:725]:
```suggestion
	future.then(|result| ready(result.or_else(|err| {
```

File: client/rpc/src/state/state_light.rs [hunk 6]
```diff
@@ -722,13 +722,10 @@ fn maybe_share_remote_request<Block: BlockT, Requests, V, IssueRequest, IssueReq
...
 			warn!("Remote request for subscription data has failed with: {:?}", err);
 			Err(())
-		},
-	}))
+		} })))
```
Comment (shawntabrizi) [inline client/rpc/src/state/state_light.rs:728]:
```suggestion
		})))
```

File: client/service/src/lib.rs [hunk 7]
```diff
@@ -392,19 +392,16 @@ fn start_rpc_servers<
...
-						warn!("Unable to bind RPC server to {}. Trying random port.", address);
-						address.set_port(0);
-						start(&address)
-					},
-					_ => Err(e),
-				})?),
-			None => None,
-		})
+		address.map(|mut address| start(&address)
+		.or_else(|e| match e.kind() {
```
Comment (shawntabrizi) [inline client/service/src/lib.rs:396]: I think this should be indented

**Review by shawntabrizi — APPROVED** [review]

**Comment by shawntabrizi** [comment]
Probably can sanity check any `{ {` presence in the codebase

**Review by apopiak — APPROVED** [review]
LGTM

**Comment by shawntabrizi** [comment]
bot merge

**Pull request merged**